pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -17,22 +17,56 @@ limitations under the License.
 package util
 
 import (
+	"sync"
+
 	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-// GetNamespacesFromPodAffinityTerm returns a set of names
-// according to the namespaces indicated in podAffinityTerm.
-// If namespaces is empty it considers the given pod's namespace.
+// NamespaceLister helps list Namespaces.
+type namespaceLister interface {
+	// List lists all Namespaces in the indexer.
+	List(selector labels.Selector) (ret []*v1.Namespace, err error)
+}
+
+var (
+	once     sync.Once
+	nsLister namespaceLister
+)
+
+// InitNamespaceLister initialize the namespace lister
+func InitNamespaceLister(namespaceListerLister namespaceLister) {
```

Comment:
this as a global is a code smell. if namespace info is required for this predicate, the method for obtaining it should be communicated from the callers.

Reply:
Thanks for point it out, will update to plumb the namespace lister to all the callers.